### PR TITLE
Test a few React components

### PR DESCRIPTION
**Pull request by robintown**
Since so much of this code base is made up React components, we do need to test them, and I wanted to see how easy it would be to get some automated accessibility tests with [axe](https://github.com/dequelabs/axe-core/) up and running. Along the way I discovered the reason why we were seeing funny errors in our tests (it was because I had turned off Vitest's test isolation option), and the tests I wrote also caught a [real accessibility issue](https://github.com/element-hq/compound-web/pull/246) in Compound Web.

### CODE REVIEW

File: vitest.config.js [hunk 1]
```diff
@@ -12,7 +12,6 @@ export default defineConfig((configEnv) =>
             classNameStrategy: "non-scoped",
           },
         },
-        isolate: false,
```
Comment (toger5) [inline vitest.config.js:15]: That very much explains what I have observed.

File: src/Modal.tsx [hunk 2]
```diff
@@ -98,6 +98,7 @@ export const Modal: FC<Props> = ({
               styles.drawer,
               { [styles.tabbed]: tabbed },
             )}
+            aria-describedby={undefined}
```
Comment (toger5) [inline src/Modal.tsx:94]: Why do we need this?

File: src/tile/SpotlightTile.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          maximised={false}
+          expanded={false}
+          onToggleExpanded={() => {}}
+          showIndicators
+        />,
+      );
+      expect(await axe(container)).toHaveNoViolations();
+      // Name should be visible
+      screen.getByText("Alice");
+    },
```
Comment (toger5) [inline src/tile/SpotlightTile.test.tsx]: We could add a lot more expectations here:
 - avatar url
 - expanded
 - indicators
 ...

File: src/utils/test.ts [hunk 4]
```diff
@@ -58,3 +65,67 @@ export function withTestScheduler(
     }),
   );
 }
+
+export async function withLocalMedia(
```
Comment (toger5) [inline src/utils/test.ts:83]: Would it make sense to not require tests (ignore test utils in codecov) on our test utils?

File: src/tile/SpotlightTile.test.tsx (hunk 3 above)
Comment (robintown) [reply src/tile/SpotlightTile.test.tsx]: Right, I've added some more expectations now. I wasn't able to test the specific avatar URL, as that requires a full Matrix client setup, and the indicators are (intentionally, IIRC?) excluded from the accessibility tree. But, there were still some useful things to test, and I actually caught a couple more accessibility issues this way!

**Pull request merged**